File: php/features.php
<?php
declare(strict_types=1);

// MemesioContentCreation SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class MemesioContentCreationFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new MemesioContentCreationBaseFeature();
            case "test":
                return new MemesioContentCreationTestFeature();
            default:
                return new MemesioContentCreationBaseFeature();
        }
    }
}

File: py/features.py
# MemesioContentCreation SDK feature factory

from feature.base_feature import MemesioContentCreationBaseFeature
from feature.test_feature import MemesioContentCreationTestFeature


def _make_feature(name):
    features = {
        "base": lambda: MemesioContentCreationBaseFeature(),
        "test": lambda: MemesioContentCreationTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
